### ISSUE TITLE
No NO_CHILD from Basic01 when tested domain does not exist

### ISSUE THREAD

**Issue by matsduf** [zonemaster/zonemaster-engine#744]
Testing for xn--test0.se that does not exist (and cannot exist). Note that there is no **NO_CHILD** message on level _ERROR_ as expected.

```
Seconds Level     Module       Message
======= ========= ============ =======
   0.00 INFO      SYSTEM       Using version v3.1.1 of the Zonemaster engine.
   0.09 INFO      BASIC        Parent domain 'se' was found for the tested domain.
   1.59 CRITICAL  BASIC        No NS records for tested zone from parent. NS tests skipped.
```
The .se zone returns NXDOMAIN for that non-existent domain:
```
$ dig @x.ns.se xn--test0.se +mult

; <<>> DiG 9.11.3-1ubuntu1.12-Ubuntu <<>> @x.ns.se xn--test0.se +mult
; (2 servers found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NXDOMAIN, id: 26545
;; flags: qr aa rd; QUERY: 1, ANSWER: 0, AUTHORITY: 1, ADDITIONAL: 1
;; WARNING: recursion requested but not available

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
;; QUESTION SECTION:
;xn--test0.se.		IN A

;; AUTHORITY SECTION:
se.			172800 IN SOA catcher-in-the-rye.nic.se. registry-default.nic.se. (
				2020052913 ; serial
				1800       ; refresh (30 minutes)
				1800       ; retry (30 minutes)
				864000     ; expire (1 week 3 days)
				7200       ; minimum (2 hours)
				)

;; Query time: 0 msec
;; SERVER: 2001:67c:124c:e000::4#53(2001:67c:124c:e000::4)
;; WHEN: Fri May 29 16:11:45 CEST 2020
;; MSG SIZE  rcvd: 117
```
[Basic01] says that **NO_CHILD** should be outputted (this is a "normal test") and that message has the level _ERROR_.

```
5. If the lookup reaches a name server that authoritatively responds
   (AA flag set) and either with NXDOMAIN for the *Child Zone* or
   with NOERROR and no record in the answer section (NODATA): 
   1. The zone returning NXDOMAIN or NODATA is defined to be the parent 
      zone. Output *[PARENT_FOUND]*.
   2. Repeat the SOA query for the *Child Zone* to the remaining name 
      servers for the parent zone.
      1. If any server returns a referral (delegation) directly to the *Child
      	 Zone*, then output *[INCONSISTENT_DELEGATION]* and go back to 
         step 4 with the found delegation.
   3. The non-existence of the *Child Zone* has been determined. 
      1. If *Test Type* is "normal test", then output *[NO_CHILD]* 
         and exit.
      2. Or, if *Test Type* is "undelegated test", then output 
         *[CHILD_NOT_DELEGATED]* and exit.

(...)

Message                        |Default severity level of message
-------------------------------|----------------------------------------------
ROOT_HAS_NO_PARENT             |INFO
PARENT_FOUND                   |INFO
PARENT_INDETERMINED            |NOTICE
CHILD_FOUND                    |INFO
NO_CHILD                       |ERROR
CHILD_NOT_DELEGATED            |INFO
INCONSISTENT_DELEGATION        |ERROR
```

[Basic01]: https://github.com/zonemaster/zonemaster/blob/v2019.2.2/docs/specifications/tests/Basic-TP/basic01.md

**Comment by vlevigneron** [comment]
Directly related to zonemaster/zonemaster-engine#568 issue which is not yet implemented

**Comment by matsduf** [comment]
This issue assumed that #568 was implemented, and will be resolved when #568 is implemented. This issue is not needed.
